x/margin/keeper/msg_server.go

```diff
@@ -328,6 +326,31 @@ func (k Keeper) ForceCloseLong(ctx sdk.Context, msg *types.MsgForceClose, isAdmi
 	return &mtp, repayAmount, nil
 }
 
+func (k msgServer) AdminClose(goCtx context.Context, msg *types.MsgAdminClose) (*types.MsgAdminCloseResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	signer, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, err
+	}
+	if !k.AdminKeeper().IsAdminAccount(ctx, admintypes.AdminType_MARGIN, signer) {
+		return nil, sdkerrors.Wrap(admintypes.ErrPermissionDenied, fmt.Sprintf("signer not authorised: %s", msg.Signer))
+	}
+
+	params := k.GetParams(ctx)
+	if msg.CloseAll {
+		params.SafetyFactor = sdk.MustNewDecFromStr("100.00")
+	}
+	params.ForceCloseFundPercentage = sdk.ZeroDec()
```

Comment:
@Jedi2002 
I don't think we want to set the system-wide forceclosefund percentage for closing an individual mtp. May make more sense to change the call to Repay() in forcecloselong to pass a a variable for takeInsurance instead of always true